scalyr_agent/log_processing.py

```diff
@@ -2282,6 +2337,46 @@ def config(self):
         """
         return self.__log_entry_config
 
+    def finish( self ):
+        """
+        Tells the log matcher to finish processing any processors.
+
+        LogMatchers that are `finished` no longer match log files or create new processors, unless
+        they haven't ever matched any files, in which case they will match at least once
+        """
+        self.__lock.acquire()
+        try:
+            # set the matcher as finished
+            self.__is_finished = True
+
+            # set any existing processors to close when they reach eof
+            for processor in self.__processors:
+                processor.close_at_eof()
+        finally:
+            self.__lock.release()
+
+    def is_finished( self ):
+        """
+        Returns true if the log matcher is finished and all of its processors are closed, and the matcher has
+        been processed at least once
+        """
+        self.__lock.acquire()
+        try:
+            # we are only finished if `is_finished` is set and we have tried to check for a match
+            # on the matcher at least once
+            if self.__is_finished and self.__last_check is None:
```

Comment:
No.  This is catching the case where the matcher has been marked as `finished`, but we have never tried to match it.  In those cases we need to return false, otherwise we won't pick up short lived files.  I'll reword the comment to make this more clear.